### PR TITLE
Alert-naming-changes

### PR DESCRIPTION
**Pull request by VatsalJagani**

### CODE REVIEW

**Review by hardikhdholariya — APPROVED** [review]

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 1]
```diff
@@ -5970,7 +5980,7 @@ search = `cs_linux_users` UID!=""\
 action.cyences_notable_event_action.products = Linux
 
 
-[Linux - User Added/Updated/Deleted]
+[Linux - User change]
```
[the remark this answers is not in the document]
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/savedsearches.conf]: Done

File: cyences_app_for_splunk/default/macros.conf [hunk 2]
```diff
@@ -12,7 +12,11 @@ iseval = 0
 
 [cs_filter_savedsearches(1)]
 args = name
-definition = regex savedsearch_name="($name$)\s" | rex field=savedsearch_name "^.*\-\s(?<Name>.*)$"
+definition = eval products_to_search="$name$" | makemv delim="|" products_to_search | eval products_to_search=trim(products_to_search) \
+| cyencesmvintersection set1="products_to_search" set2="alert_products" \
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/macros.conf:16]: Just to make sure values are not passed as a string to set1 and set2

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml [hunk 3]
```diff
@@ -141,30 +143,12 @@
...
-| search NOT savedsearch_name IN `cs_monthly_alerts`
-| sort savedsearch_name, cyences_severity
-| streamstats count as newcount by savedsearch_name
-| where (cyences_severity!="unknown" OR newcount=1)
-| makemv delim="," teams
-| eval teams=trim(teams)
-| search $tkn_teams$
-| `cs_filter_savedsearches("Sophos Endpoint Protection|Windows Defender|CrowdStrike|Defender ATP")` 
+        <search base="basesearch">
+          <query>| `cs_filter_savedsearches("Sophos Endpoint Protection|Windows Defender|CrowdStrike|Defender ATP")`
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/data/ui/views/cs_overview.xml:147]: cs_filter_savedsearches => this macro searching based on the product name so need to change the filters everywhere where we don't have search prefix as a product.

for ex.: line no. 126- we don't have Authentication as a product.

Comment (hardikhdholariya) [reply cyences_app_for_splunk/default/data/ui/views/cs_overview.xml:147]: same way for "Defender ATP" alerts, product is "Office 365 Defender ATP"

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml [hunk 4]
```diff
@@ -619,6 +605,34 @@
       </table>
     </panel>
   </row>
+  <row>
+    <panel> <!-- TODO - hide logic to add-->
+      <title>Other Alerts</title>
+      <table id="tbl_other_alerts">
+        <search base="basesearch">
+          <query> eval products_to_search="Authentication|VPN|Radius Authentication|Sophos Endpoint Protection|Windows Defender|CrowdStrike|Defender ATP|AWS|Google Workspace|O365|Email|Network Compromise|Cisco IOS|Cisco Meraki|Vulnerability|Windows|AD|Credential Compromise|Ransomware|Linux|F5 BIGIP|Oracle|MSSQL"
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/data/ui/views/cs_overview.xml:613]: same way product names to change as we have in cs_product_list dict

**Comment by mahirchavda** [comment]
We need to handle update scenario. Otherwise all the configured alert willl be stopped due to rename

**Review by hardikhdholariya — CHANGES_REQUESTED** [review]

**Comment by VatsalJagani** [comment]
> We need to handle update scenario. Otherwise all the configured alert willl be stopped due to rename

Actually it will be very complicated, so we'll ask user to go through App setup page instead, and disable and reenable product. As this is 5.0.0 so I think it should be okay to suggest user on doing this instead. Unless @hardikhdholariya if you have any better suggestion to handle this.

@hardikhdholariya  - We need to discuss this next week so we can discuss all points regarding 5.0.0

File: cyences_app_for_splunk/default/macros.conf (hunk 2 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/macros.conf:16]: They are field names passed to set1 and set2

That's how we are doing for mvdiff command. Not sure if there is a better way to do it.

File: cyences_app_for_splunk/appserver/static/cs_overview.js [hunk 5]
```diff
@@ -66,7 +66,7 @@ require([
         { token: 'vulnerability', associated_products: ['Qualys', 'Tenable', 'Nessus', 'CrowdStrike Spotlight'] },
         { token: 'ad_windows', associated_products: ['Sysmon', 'Windows', 'Windows AD', 'Windows DNS'] },
         { token: 'credential_compromise', associated_products: ['Sysmon'] },
-        { token: 'ransomware', associated_products: ['Sysmon', 'Windows', 'Cisco IOS', 'FortiGate', 'Palo Alto', 'Sophos Firewall', 'Cisco Meraki'] },
+        { token: 'ransomware', associated_products: ['Ransomware', 'Sysmon'] },
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/appserver/static/cs_overview.js:69]: is "TOR Traffic detected by Firewall" alert is falling under ransomware? previously, It was under ransomware. if so then we have to add following products as well.

'Windows', 'Cisco IOS', 'FortiGate', 'Palo Alto', 'Sophos Firewall'

File: cyences_app_for_splunk/appserver/static/cs_overview.js [hunk 6]
```diff
@@ -56,7 +56,7 @@ require([
     ]
 
     let panel_depends_tokens = [
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/appserver/static/cs_overview.js:58]: I think this list we don't have to change as it used in show/hide overview dashboard panel based on enable/disable products.

so no need to add Authentication, Ransomware and Sysmon tags here.

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml [hunk 7]
```diff
@@ -144,7 +144,7 @@
       <title>Antivirus</title>
       <table id="tbl_antivirus">
         <search base="basesearch">
-          <query>| `cs_filter_savedsearches("Sophos Endpoint Protection|Windows Defender|CrowdStrike|Defender ATP")`
+          <query>| `cs_filter_savedsearches("Sophos Endpoint Protection|Windows Defender|CrowdStrike|Defender ATP|Office 365 Defender ATP")`
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/data/ui/views/cs_overview.xml:147]: we can remove "Defender ATP" safely.

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml [hunk 8]
```diff
@@ -291,7 +291,7 @@
       <title>Office 365</title>
       <table id="tbl_o365">
         <search base="basesearch">
-          <query>| `cs_filter_savedsearches("O365")` 
+          <query>| `cs_filter_savedsearches("O365|Office 365")`
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/data/ui/views/cs_overview.xml:294]: we can remove "O365" safely.

File: cyences_app_for_splunk/appserver/static/cs_overview.js (hunk 5 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/appserver/static/cs_overview.js:69]: No, its not specific to Ransomware. its generic security issue on Network.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 9]
```diff
@@ -5573,7 +5575,7 @@ action.cyences_notable_event_action.system_compromised_drilldown = index=* `cs_v
 action.cyences_notable_event_action.attacker_search = | stats sum(count) as count by user
 action.cyences_notable_event_action.attacker_drilldown = index=* `cs_vpn_indexes` tag=authentication action="failure" dest_category="vpn_auth" user=$row.user$
 action.cyences_send_email_action = 1
-action.cyences_notable_event_action.products = VPN
+action.cyences_notable_event_action.products = VPN, Authentication
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/savedsearches.conf:5578]: same way we need to add the "Email" tag to following alerts:

Email - Calculate UpperBound for Spike In Emails
Email - Hourly Increase In Emails Over Baseline
Email - Daily Spam Email

Include "Vulnerability" tag for "Vulnerability - Detected Vulnerabilities" alert

Also need to check the alerts displayed under "Active Directory & Windows" panel. => may be update product name from cs_filter_savedsearches macro or update products associated to alerts.

Comment (hardikhdholariya) [reply cyences_app_for_splunk/default/savedsearches.conf:5578]: Also, for following alerts, need to add "Credential Compromise" tag

Credential Compromise - Windows - Credential Dumping through LSASS Access
Credential Compromise - Windows - Credential Dumping via Symlink to Shadow Copy
Credential Compromise - Windows - Credential Dumping via Copy Command from Shadow Copy
Credential Compromise - Windows - Credential Dump From Registry via Reg exe

File: cyences_app_for_splunk/bin/cyences_mvintersection.py [hunk 10]
```diff
@@ -0,0 +1,44 @@
+import sys
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/bin/cyences_mvintersection.py:1]: Just add **import cs_imports** for the splunklib import

File: cyences_app_for_splunk/appserver/static/cs_overview.js (hunk 6 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/appserver/static/cs_overview.js:58]: Let's keep them I think just as fail safe.

I think Sysmon is needed as Sysmon we have it as product in the product list.

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml (hunk 7 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/data/ui/views/cs_overview.xml:147]: Done

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml (hunk 3 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/data/ui/views/cs_overview.xml]: Done

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml (hunk 4 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/data/ui/views/cs_overview.xml]: Done

File: cyences_app_for_splunk/default/data/ui/views/cs_overview.xml (hunk 8 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/data/ui/views/cs_overview.xml:294]: done

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 9 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/default/savedsearches.conf:5578]: Added Email tag, Vulnerability tag and Network tag to the genertic alerts in that category.
Please let me know if any other generic category is remaining.

Comment (VatsalJagani) [reply cyences_app_for_splunk/default/savedsearches.conf:5578]: Windows AD tag fixed.

Comment (VatsalJagani) [reply cyences_app_for_splunk/default/savedsearches.conf:5578]: FYI, no need to add Credential Compromise tag as its not a product type or data source type.

File: cyences_app_for_splunk/bin/cyences_mvintersection.py (hunk 10 above)
Comment (VatsalJagani) [reply cyences_app_for_splunk/bin/cyences_mvintersection.py:1]: Done

**Review by hardikhdholariya — APPROVED** [review]

**Pull request merged**